nengo/connection.py

```diff
@@ -233,9 +272,12 @@ class Connection(NengoObject):
     synapse : Synapse, optional \
               (Default: ``nengo.synapses.Lowpass(tau=0.005)``)
         Synapse model to use for filtering (see `~nengo.synapses.Synapse`).
-    function : callable, optional (Default: None)
+    function : callable or (n_eval_points, size_mid) array_like, \
+               optional (Default: None)
         Function to compute across the connection. Note that ``pre`` must be
         an ensemble to apply a function across the connection.
+        If an array is passed, the function is implicitly defined by the
+        array points and the provided ``eval_points``.
```

Comment:
> It gives the required shape above. Does that make it clear that it's 2D?

Aw jeez, I totally missed that notation.

> I think the assumption is that anything passed as a parameter is not copied

That's true. I forgot about that assumption when it comes to Python.